src/main/java/bio/terra/stairway/FlightContext.java

```diff
@@ -104,6 +105,18 @@ public FlightMap getWorkingMap() {
     return workingMap;
   }
 
+  /**
+   * Replace the default map created at constructor time.
+   *
+   * @param map If not empty, replaces the existing workingMap with the contained map. If empty,
+   *     existing map is left as-is.
+   */
+  public void setWorkingMap(Optional<FlightMap> map) {
```

Comment:
tiny nit: taking an optional as an argument is a little funny to me here. Maybe just take `FlightMap` and force the client to only call the function if they have something to set?

Reply:
Reverted to existing code and did check external to FlightContext.